cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -2862,6 +2863,47 @@ deserialiseFromBech32 asType bech32Str = do
 
     return value
 
+deserialiseAnyOfFromBech32
+  :: forall b.
+     [FromSomeType SerialiseAsBech32 b]
+  -> Text
+  -> Either Bech32DecodeError b
+deserialiseAnyOfFromBech32 types bech32Str = do
+    (prefix, dataPart) <- Bech32.decodeLenient bech32Str
+                            ?!. Bech32DecodingError
+
+    let actualPrefix = Bech32.humanReadablePartToText prefix
+
+    FromSomeType actualType fromType <-
+      findForPrefix actualPrefix
+        ?! Bech32UnexpectedPrefix actualPrefix permittedPrefixes
+
+    payload <- Bech32.dataPartToBytes dataPart
+                 ?! Bech32DataPartToBytesError (Bech32.dataPartToText dataPart)
+
+    value <- deserialiseFromRawBytes actualType payload
+               ?! Bech32DeserialiseFromBytesError payload
+
+    let expectedPrefix = bech32PrefixFor value
+    guard (actualPrefix == expectedPrefix)
+      ?! Bech32WrongPrefix actualPrefix expectedPrefix
+
+    return (fromType value)
+  where
+    findForPrefix
+      :: Text
+      -> Maybe (FromSomeType SerialiseAsBech32 b)
+    findForPrefix prefix =
+      find
+        (\(FromSomeType t _) -> prefix `elem` bech32PrefixesPermitted t)
```

Comment:
Where is `bech32PrefixesPermitted` coming from?

Reply:
`bech32PrefixesPermitted` is a function of the `SerialiseAsBech32` typeclass